src/backend/core/models.py

```diff
@@ -284,8 +284,27 @@ def __str__(self):
         return f"{id_str:s}{main_str:s}"
 
     def save(self, *args, **kwargs):
-        """Ensure users always have one and only one main identity."""
+        """
+        Save identity, ensuring user always have one and only one main identity.
+        Also convert valid invitations if new identity.
+        """
+
+        # If new identity, convert all valid invitations to team accesses.
+        # Expired invitations are ignored.
+        if self._state.adding is True:
+            validity_duration = timedelta(seconds=settings.INVITATION_VALIDITY_DURATION)
+            active_invitations = Invitation.objects.filter(
+                email=self.email, created_at__gte=timezone.now() - validity_duration
+            )
+            for invitation in active_invitations:
+                TeamAccess.objects.create(
+                    user=self.user, team=invitation.team, role=invitation.role
+                )
+                invitation.delete()
```

Comment:
Have you though using a `bulk_create()`? [doc](https://docs.djangoproject.com/en/4.1/ref/models/querysets/#bulk-create).

It would save us at least n queries. However, I think bulk delete doesn't exist, we might still need to call it n-times.
Wdyt?

Reply:
I hadn't ! I've looked into "bulk_delete" solutions but none of them seem satisfying to me. 

Would you use `bulk_create` if we still have to use a for loop for deletion ?

Reply:
do you have any input @sampaccoud?